### PR TITLE
AB#91040 - adjusted home type screens and added validation

### PR DESCRIPTION
**Pull request by rafalgajewski**

### CODE REVIEW

File: src/HE.Investment.AHP.Domain/HomeTypes/CommandHandlers/SaveFinishHomeTypesAnswerCommandHandler.cs [hunk 1]
```diff
@@ -14,17 +15,29 @@ public class SaveFinishHomeTypesAnswerCommandHandler : HomeTypeCommandHandlerBas
...
         var account = await _accountUserContext.GetSelectedAccount();
         var homeTypes = await _repository.GetByApplicationId(request.ApplicationId, account, HomeTypeSegmentTypes.None, cancellationToken);
+        var scheme = await _schemeRepository.GetByApplicationId(request.ApplicationId, account, false, cancellationToken);
+
+        if (request.FinishHomeTypesAnswer == Contract.HomeTypes.Enums.FinishHomeTypesAnswer.Yes
+            && !request.IsCheckOnly
+            && scheme?.Funding?.HousesToDeliver != homeTypes.HomeTypes.Sum(x => x.HomeInformation?.NumberOfHomes?.Value))
+        {
+            return OperationResult.New().AddValidationError("HomeTypes", "You have not assigned all of the homes you are delivering to a home type");
+        }
```
Comment (jjankowski87) [inline src/HE.Investment.AHP.Domain/HomeTypes/CommandHandlers/SaveFinishHomeTypesAnswerCommandHandler.cs:39]: This logic should not be in command handler but in CompleteSection method in HomeTypes entity.
ApplicationBasicInfo class is already injected into HomeTypes entity so one of possibility is to extend that info with number of Houses to deliver (and fetch it with single request to CRM)

**Review by jjankowski87 — COMMENTED** [review]
You have not updated Finish home types page

File: src/HE.Investment.AHP.Domain/HomeTypes/CommandHandlers/SaveFinishHomeTypesAnswerCommandHandler.cs (hunk 1 above)
Comment (rafalgajewski) [reply src/HE.Investment.AHP.Domain/HomeTypes/CommandHandlers/SaveFinishHomeTypesAnswerCommandHandler.cs]: moved logic to entity

**Review by jjankowski87 — APPROVED** [review]

File: src/HE.Investment.AHP.Domain/HomeTypes/Entities/HomeTypesEntity.cs [hunk 2]
```diff
@@ -119,6 +121,11 @@ public void CompleteSection(FinishHomeTypesAnswer finishAnswer)
                     notCompletedHomeTypes.Select(x => new ErrorItem($"HomeType-{x.Id}", $"Complete {x.Name.Value} to save and continue")).ToList()));
             }
 
+            if (expectedNumberOfHomes != _homeTypes.Sum(x => x.HomeInformation?.NumberOfHomes?.Value ?? 0))
```
Comment (jjankowski87) [inline src/HE.Investment.AHP.Domain/HomeTypes/Entities/HomeTypesEntity.cs:124]: Please run Intergration tests - number of homes is hardcoded in Home Types and dynamic (random) in scheme. I would go with hardcoding value in scheme section.

**Review by Antek012 — APPROVED** [review]

File: src/HE.Investment.AHP.Domain/HomeTypes/Entities/HomeTypesEntity.cs (hunk 2 above)
Comment (rafalgajewski) [reply src/HE.Investment.AHP.Domain/HomeTypes/Entities/HomeTypesEntity.cs:124]: Integration test fixed.

**Pull request merged**